src/core.jl

```diff
@@ -1370,15 +1370,37 @@ Base.haskey(graph::Graph, name) = isnull(get_node_by_name(graph, name))
 
 
 
+node_name(::Void) = nothing
+node_name(xs::AbstractVector)=node_name.(xs)
```

Comment:
Couldn't you just call `node_name.(x)` whereever you expect an array? On 0.6 at least that should behave the right way for a single tensor.